llvm/lib/Transforms/Instrumentation/InstrProfiling.cpp

```diff
@@ -930,23 +930,8 @@ Value *InstrLowerer::getCounterAddress(InstrProfCntrInstBase *I) {
   LoadInst *&BiasLI = FunctionToProfileBiasMap[Fn];
   if (!BiasLI) {
```

Comment:
@ellishg I've confirmed it works fine if `FunctionToProfileBiasMap` is eliminated and make this load in-place, since redundant `BiasLI` is optimized out.
Do you think this shall be preserved? (6e23cd2bf073ab5eb5a103b935af54046d007c52)

Reply:
I think as long as you don't rely on `Fn->getEntryBlock().front()` it should be fine.

Reply:
I'll do in the next time. I have to reconfirm the effort of "counter promotion".

Reply:
@ellishg I meant, "I'll do in the next opportunity."

I tried sinking LoadLI from the entry to each used block. It made things worse since they weren't optimized out in larger functions.

For now, I'll leave LoadLI untouched. I have to investigate and re-evaluate register spills.